lib/action_view/template/handlers/markdown.rb

```diff
@@ -0,0 +1,29 @@
+module ActionView
+  module Template::Handlers
+    class Markdown
+      def call(_template, source)
+        @markdown ||= Redcarpet::Markdown.new(Redcarpet::Render::HTML.new(with_toc_data: true))
+        "
+        content_for :html_list_of_headings_links do
+          #{wrap_in_a_div(rendered_table_of_contents_for(source)).inspect}.html_safe
```

Comment:
This line looks really awkward and fragile..... but I admit I'm struggling to find a better alternative that works :/

Reply:
Yeah it's awkward because it's generating ruby code. I've tried to provide lots of spec coverage to ensure it keeps working.